Feat: 커뮤니티 글 API 구현

gachaemin opened the pull request:
1. 커뮤니티 글 작성 API 구현
2. 커뮤니티 글 수정 API 구현
3. 커뮤니티 글 삭제 API 구현

## ✅ PR 유형
어떤 변경 사항이 있었나요?

- [x] 새로운 기능 추가
- [ ] 버그 수정
- [ ] 리팩토링
- [ ] 코드에 영향을 주지 않는 변경사항(주석, 개행 등등..)
- [ ] 문서 수정
- [ ] 빌드 부분 혹은 패키지 매니저 수정
- [ ] 테스트 코드 추가

---

## ✏️ 작업 내용
![image](https://github.com/user-attachments/assets/4f799299-a7a3-479f-901b-e9b6c2aa0e9c)
요청을 받는 PostController 파일입니다.

![image](https://github.com/user-attachments/assets/150c82b8-9b7b-4078-b003-3cf3d823790b)
비즈니스 로직 분리를 위한 PostService 파일입니다.

![image](https://github.com/user-attachments/assets/36522696-7d17-46e6-b94b-1f7d20b595c9)
dto파일의 구조입니다.

![image](https://github.com/user-attachments/assets/46e5afd1-3cfb-434c-9ea2-03b36cdacdd8)
![image](https://github.com/user-attachments/assets/f11b0da1-9c04-4072-8033-526fc0d065f3)
각각 Request/Response 파일의 예시입니다.

![image](https://github.com/user-attachments/assets/220f5ec1-7cf8-4467-b4ad-5cd82ca659a3)
API 문서화를 위한 Swagger 명세 인터페이스 코드입니다.

---

## 🔗 관련 이슈
- closes #22 

---

## 💡 추가 사항

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **신규 기능**
  - 게시글 생성, 수정, 삭제를 위한 REST API 엔드포인트가 추가되었습니다.
  - 게시글 생성, 삭제, 수정 요청 및 응답 데이터 구조가 도입되었습니다.
  - JWT 기반 인증 및 권한 관리 기능이 도입되어 보안이 강화되었습니다.
  - CORS 설정이 추가되어 특정 도메인에서의 API 접근이 허용됩니다.
  - WebSocket 및 보안 관련 의존성이 추가되었습니다.

- **문서화**
  - 게시글 관련 API에 대한 Swagger(OpenAPI) 문서가 추가되어 요청 및 응답 예시와 설명이 제공됩니다.

- **버그 수정 및 개선**
  - Redis 자동 구성 기능이 비활성화되어 불필요한 리소스 사용이 줄어들었습니다.
  - 예외 처리 로직이 개선되어 다양한 오류 상황에 대해 일관된 응답과 로깅이 제공됩니다.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

gachaemin committed:
Feat: 커뮤니티 글 API 구현
1. 커뮤니티 글 작성 API 구현
2. 커뮤니티 글 수정 API 구현
3. 커뮤니티 글 삭제 API 구현

han16935 commented on src/main/java/com/playus/communityservice/domain/post/controller/PostController.java:
게시글 생성 후 게시글의 pk를 프론트 쪽으로 넘겨주는 response가 잇으면 좋을 거 같아요~

han16935 replied:
게시글 업데이트, 삭제 후 프론트 측에 성공했다는 메시지를 넘겨주는 건 어떨까요? 예를 들면 "게시글 업데이트에 성공했습니다!" "게시글 삭제 성공했습니다!" 요런 느낌으로요 보니까 community 측 API 명세서 response에 아무것도 없던데 굳이 저걸 다 지킬 필요는 없습니다! 필요, 혹은 피그마 디자인 보면서 바꾸셔도 되요!

gachaemin replied:
API명세서 그대로 진행해야하는줄 알았어요..!! response를 추가하도록 할게요!

han16935 commented on src/main/java/com/playus/communityservice/domain/post/service/PostService.java:
try-catch 대신 `ControllerAdvice` 라는 걸 써서 예외 처리를 해 보시는 건 어떨까요? 이번 pr에선 안 보여서 혹시 이미 관련 클래스가 있다면 신경 안 쓰셔도 됩니다...!

gachaemin replied:
try-catch보다 controllerAdvice를 추천하시는 이유를 알 수 있을까요?

han16935 replied:
`try-catch` 문 같은 경우는 지저분하기도 하고, business logic을 처리하는 곳과 exception 을 처리하는 곳은 분리되어야 보다 관심사의 분리가 이뤄진다고 생각해요~ 많이 사용되는 곳은 예를 들면 앞서 request 객체에 있는 `@NotNull` 같은 경우에도 
따로 작성하신 규칙을 위반하면 `BindException` 이 throw 되는데, throw 될 경우에 따로 controller 등에 관련 로직을 작성할 필요 없이, advice에서 처리가 가능해요!
```
@ResponseStatus(HttpStatus.BAD_REQUEST)
    @ExceptionHandler(BindException.class)
    public ErrorResponse bindExceptionHandler(BindException e) {
        String errorMessage = e.getAllErrors().get(0).getDefaultMessage();
        log.warn("Validation Error: {}", errorMessage);
        return ErrorResponse.badRequestError(errorMessage);
    }
```

han16935 commented on src/main/java/com/playus/communityservice/domain/post/service/PostService.java:
Custom Exception을 쓰지 않고 `IllegalArgumentException` 통해 처리하신 이유가 잇을까요?

gachaemin replied:
간단한 개발이 가능하다고 알아서 적용했습니다. 그런데 검색해보니 정확한 경고문이 뜨는게 아니더라구요. 다른 내용 변경하면서 이것도 따로 변경하도록 하겠습니다.

han16935 replied:
넵! custom exception과 표준 예외 대해서는 무조건 이쪽이 맞다! 그 정도까진 아니여서 한 번 찾아보셔도 괜찮을 거 같아요~

han16935 reviewed (changes requested):
고생하셧습니다~ pr 제목을 구체적으로 어떤 API를 구현했는지 작성해주시는 것도 괜찮을 거 같아요! 테스트는 어떻게 진행하셧나요? 아 그리고 rabbit이 댓글 달아준 건 전부 답변해 주시고 대략 이해햇습니다~ 이렇게 납득한 거 같은 답변이 날아오면 resolve 해 주세요!

gachaemin commented on src/main/java/com/playus/communityservice/domain/post/dto/post_create/PostCreateRequest.java:
```diff
@@ -0,0 +1,29 @@
+
+        @NotBlank(message = "게시글 제목이 비어 있습니다!")
+        @Size(max = 225, message = "제목의 길이를 1~225자 이내로 작성해 주세요!")
+        String title,
```
@coderabbitai
이해했습니다! 변경하도록 하겠습니다

han16935 commented on src/main/java/com/playus/communityservice/CommunityServiceApplication.java:
여기서 Redis 관련 configuration 클래스를 exclude 하신 이유가 있을까요?

gachaemin replied:
community에서는 redis를 사용하지 않는걸로 알고 있고, redis관련 오류가 발생하여 exclude하였습니다! 혹시 나중에 필요하다고 하면 다시 추가하려고 합니다.

han16935 commented on src/main/java/com/playus/communityservice/domain/post/service/PostService.java:
`@Transactional` 을 method 단이 아닌 class 단에 작성하셔도 괜찮을 거 같아요!

gachaemin replied:
일단 따로 했었는데, 저도 확인해보니 class단으로 작성하는게 더 깔끔할거같네요. 변경하도록 하겠습니다!

han16935 commented on src/main/resources/application-local.yaml:
여기서 `config.import` 이 부분은 제가 한 번도 안 써봐서 그러는데 실례가 안 된다면 혹시 설명 부탁드려도 괜찮을까요...?

gachaemin replied:
.env파일을 환경변수에 경로로 적용시켰는데 제꺼에서는 자꾸 env파일 인식을 못해서, local에 config.import를 사용해서 env파일을 인식하도록 설정했습니다!

gachaemin commented on src/main/java/com/playus/communityservice/domain/post/specification/PostControllerSpecification.java:
```diff
@@ -0,0 +1,88 @@
+                    )
+            )
+    })
+    ResponseEntity<String> createPost(PostCreateRequest request);
```
@coderabbitai
코드 작업에만 집중하여 빠트린것으로 확인됩니다! 다음 commit코드에서 변경하도록 하겠습니다.

gachaemin commented on src/main/java/com/playus/communityservice/domain/post/service/PostService.java:
```diff
@@ -0,0 +1,86 @@
+    public void createPost(PostCreateRequest request) {
+        // 기본값 지정
+        Long writerId = 1L; // 나중에 인증 연동 시 대체
+        Tag defaultTag = Tag.LG_TWINS; // 기본 태그 (클라이언트가 안 보내니까 서버에서 결정)
```
@coderabbitai
이 부분은 제가 정확하게 알지 못해서, 팀원들과 이야기하고 변경하도록 하겠습니다.

yechan-kim replied:
작성자 ID 같은 경우에는 컨트롤러에 `@AuthenticationPrincipal`을 붙여, JWT 토큰에서 추출한 인증 객체를 통해 작성자 ID를 저장할 수 있습니다.
또한 `Tag` 같은 경우에는, 클라이언트에서 제공하는 방법으로 변경하는 것이 좋을 것 같습니다! 해당 부분은 클라이언트 파트에게 전달하도록 하겠습니다.

han16935 commented on src/main/java/com/playus/communityservice/domain/post/controller/PostController.java:
아 그리고 createPost() 반환 시 200 대신 201로 반환해주세요! ResponseStatus로 하실 거면 `ResponseStatus.created().body(xxx);` 느낌으로 하면 됐을 거에요!

gachaemin replied:
controller 수정하면서 적용하도록 하겠습니다!

gachaemin commented on src/main/java/com/playus/communityservice/domain/post/specification/PostControllerSpecification.java:
```diff
@@ -0,0 +1,90 @@
+                    )
+            )
+    })
+    ResponseEntity<String> deletePost(PostDeleteRequest request);
```
@coderabbitai
전체적으로 response 파일을 제작하여 모든 타입을 통일하도록 변경할 예정입니다.

gachaemin commented on src/main/java/com/playus/communityservice/domain/post/service/PostService.java:
```diff
@@ -0,0 +1,94 @@
+        Long currentUserId = 1L;
+        if (!post.getWriterId().equals(currentUserId)) {
+            throw new IllegalStateException("게시글 작성자만 삭제할 수 있습니다.");
+        }
```
@coderabbitai
TeamTag와 연동하여 하드코딩없이 사용하도록 수정하겠습니다.

gachaemin committed:
Feat: 전체적인 코드의 기능 추가
1. JWT관련 코드 추가
- JwtUtil과 JwtFilter를 추가하여 인증 활성화
- JwtUser를 추가하여 작설자 ID 추가
2. Security 관련 코드 추가
3. Service에 ID와 TAG를 연동하여 하드코딩 제거
4. Exception의 처리를 위한 코드 추가
5. Exception에서 사용하는 ErrorResponse 추가
6. Response 추가
- PostUpdateResponse 추가
- PostDeleteResponse 추가
7. 직관 일지를 위한 LocalDate 추가

gachaemin commented on src/main/java/com/playus/communityservice/domain/post/dto/post_create/PostCreateResponse.java:
```diff
@@ -0,0 +1,17 @@
+        String massage
+) {
+
+    public static PostCreateResponse of(Long postId, String massage) {
+        return PostCreateResponse.builder()
+                .postId(postId)
+                .massage("게시물 생성이 완료되었습니다.")
+                .build();
+    }
```
오타 수정 후, 파라미터 제거하도록 하겠습니다

gachaemin replied:
@coderabbitai
오타 수정 후 파라미터 제거하도록 하겠습니다.

gachaemin commented on src/main/java/com/playus/communityservice/domain/post/dto/post_create/PostCreateResponse.java:
```diff
@@ -0,0 +1,17 @@
+@Builder
+public record PostCreateResponse(
+        Long postId,
+        String massage
```
@coderabbitai
오타 수정하도록 하겠습니다.

gachaemin commented on src/main/java/com/playus/communityservice/domain/post/dto/post_update/PostUpdateResponse.java:
```diff
@@ -0,0 +1,16 @@
+        boolean success,
+        String message
+) {
+    public static PostUpdateResponse fo(boolean success, String message) {
+        return PostUpdateResponse.builder()
+                .success(true)
+                .message("게시물이 수정되었습니다.")
+                .build();
+    }
```
@coderabbitai
코드 수정 후 매개 변수 수정하도록 하겠습니다.

gachaemin commented on src/main/java/com/playus/communityservice/domain/post/dto/post_delete/PostDeleteResponse.java:
```diff
@@ -0,0 +1,16 @@
+        boolean success,
+        String message
+) {
+    public static PostDeleteResponse of(boolean success, String message) {
+        return PostDeleteResponse.builder()
+                .success(true)
+                .message("게시물이 삭제되었습니다.")
+                .build();
+    }
```
@coderabbitai
수정하도록 하겠습니다.

gachaemin commented on src/main/java/com/playus/communityservice/domain/post/specification/PostControllerSpecification.java:
```diff
@@ -0,0 +1,145 @@
+                    )
+            )
+    )
+    @ApiResponses(value = {
+            @ApiResponse(
+                    responseCode = "200", description = "생성 성공",
+                    content = @Content(
+                            mediaType = APPLICATION_JSON_VALUE,
+                            examples = @ExampleObject(
+                                    name = "게시글 생성 응답 예시",
+                                    value = """
+                                    {
+                                      "postId" : 1,
+                                      "message" : "게시물 생성이 완료되었습니다."
+                                    }
+                                    """
+                            )
+                    )
+            )
+    })
```
@coderabbitai
controller 수정 후 specification을 까먹었습니다. 바로 수정하겠습니다.

gachaemin commented on src/main/java/com/playus/communityservice/domain/post/specification/PostControllerSpecification.java:
```diff
@@ -0,0 +1,145 @@
+                                      "title": "오늘 경기 재밌었어요!",
+                                      "image": "2",
+                                      "content": "긴장하면서 시청했네요."
+                                      "postId": 1
+                                      "jwpDate" : "2025-05-01"
+                                    }
```
@coderabbitai
바로 반영하겠습니다.

gachaemin commented on src/main/java/com/playus/communityservice/domain/post/enums/TeamTag.java:
```diff
@@ -0,0 +1,28 @@
+
+    private final String name;
+
+    public TeamTag fromString(String name){
+        for (TeamTag teamTag : TeamTag.values()) {
+            if (teamTag.name.equals(name)) {
+                return teamTag;
+            }
+        }
+        throw new IllegalArgumentException("No enum constant " + name);
+    }
```
@coderabbitai
바로 수정하겠습니다.

gachaemin committed:
Fix: 오타 및 메소드 변경
- 오탈자 수정
- TeamTag의 메소드 변경
- 상태가 불일치한 코드 수정

gachaemin commented on src/main/java/com/playus/communityservice/domain/post/specification/PostControllerSpecification.java:
```diff
@@ -0,0 +1,147 @@
+                                    name = "수정 응답 예시",
+                                    value = """
+                                    {
+                                        "success" : "true",
+                                        "message" : "게시물이 수정되었습니다."
```
@coderabbitai
수정하겠습니다.

gachaemin commented on src/main/java/com/playus/communityservice/domain/post/specification/PostControllerSpecification.java:
```diff
@@ -0,0 +1,147 @@
+                                    name = "삭제 응답 예시",
+                                    value = """
+                                    {
+                                        "success" : "true",
+                                        "message" : "게시물이 삭제되었습니다."
```
@coderabbitai
수정하겠습니다.

gachaemin committed:
Style: 큰따옴표 제거

Minminminwoo commented on src/main/java/com/playus/communityservice/global/config/jwt/JwtFilter.java:
```diff
@@ -0,0 +1,57 @@
+    protected void doFilterInternal(HttpServletRequest request,
+                                    HttpServletResponse response,
+                                    FilterChain chain)
+            throws ServletException, IOException {
```
프론트와 확인하다보니 엑세스 토큰을 AUTHORIZATION HttpHeader 가 아닌 쿠키로 전달하기로 방식이 바뀌었습니다. user 레포지토리에서 닉네임 중복 방지/프로필 수정하는 api 작성 후 pr 되면 그때 적용하시면 좋을것 같습니다.

gachaemin replied:
그럼 이 부분은 일단 놔둔 상태로 merge하고 나중에 변경하도록 하겠습니다.

Minminminwoo commented on src/main/java/com/playus/communityservice/domain/post/specification/PostControllerSpecification.java:
```diff
@@ -0,0 +1,147 @@
+    @Tag(name = "Post", description = "게시글 생성 API")
+    @Operation(
+            summary = "게시글 생성",
+            description = "게시글을 작성합니다.",
```
인증된 유저인지 판별을 위해 요청 보낼때 헤더에 엑세스 토큰을 담아보내줘야 해서, swagger 코드 description = "게시글을 작성합니다.", 밑에 security    = @SecurityRequirement(name = "AccessCookie"),
            parameters  = @Parameter(
                    name        = "Access",
                    description = "JWT access token (쿠키)",
                    in          = ParameterIn.COOKIE,
                    required    = true,
                    example     = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9..."
            ), 이 security와 parameters 설정도 넣어주시면 감사하겠습니다!

gachaemin replied:
바로 수정하도록 하겠습니다!

gachaemin committed:
Docs: Specification문서 수정
- security와 parameter 설정 추가

gachaemin commented on src/main/java/com/playus/communityservice/domain/post/specification/PostControllerSpecification.java:
```diff
@@ -0,0 +1,145 @@
+                    )
+            )
+    })
+    ResponseEntity<PostCreateResponse> createPost(PostCreateRequest request);
```
@coderabbitai
위와 같이 수정하겠습니다.

gachaemin commented on src/main/java/com/playus/communityservice/domain/post/specification/PostControllerSpecification.java:
```diff
@@ -0,0 +1,145 @@
+                                      "title": "오늘 경기 재밌었어요!",
+                                      "image": "1",
+                                      "content": "LG가 이길 줄 알았죠!"
+                                      "jwpDate" : "2025-05-05"
+                                    }
```
@coderabbitai
위와 같이 수정하겠습니다.

gachaemin commented on src/main/java/com/playus/communityservice/domain/post/specification/PostControllerSpecification.java:
```diff
@@ -0,0 +1,145 @@
+                    )
+            )
+    })
+    ResponseEntity<PostUpdateResponse> updatePost(PostUpdateRequest request);
```
@coderabbitai
위와 같이 수정하겠습니다.

gachaemin commented on src/main/java/com/playus/communityservice/domain/post/dto/post_create/PostCreateRequest.java:
```diff
@@ -0,0 +1,29 @@
+
+        @NotBlank(message = "본문이 비어 있습니다!")
+        @Size(max = 500, message = "본문의 길이를 1~500자 이내로 작성해 주세요!")
+        String message
```
@coderabbitai
수정하겠습니다.

gachaemin commented on src/main/java/com/playus/communityservice/domain/post/specification/PostControllerSpecification.java:
```diff
@@ -0,0 +1,88 @@
+                                    value = """
+                                    {
+                                      "ok"
+                                    }
```
@coderabbitai
수정하겠습니다.

gachaemin commented on src/main/java/com/playus/communityservice/domain/post/controller/PostController.java:
```diff
@@ -0,0 +1,38 @@
+    public ResponseEntity<String> createPost(@RequestBody @Valid PostCreateRequest request) {
+        postService.createPost(request);
+        return ResponseEntity.ok("{\"ok\"}");
+    }
```
@coderabbitai
다른 형태를 사용하여 변경하였습니다!

gachaemin commented on src/main/java/com/playus/communityservice/domain/post/service/PostService.java:
```diff
@@ -0,0 +1,86 @@
+                writerId,
+                request.title(),
+                request.message(),
+                defaultTag
```
@coderabbitai
수정하겠습니다.

han16935 commented on src/main/java/com/playus/communityservice/domain/post/controller/PostController.java:
`@PathVariable` 로 받아오는 `tag` 대해서는 어떻게 validate 되나요?

gachaemin replied:
요청된 URL의 String값이 Enum에 있는 teamName과 일치할 경우 자동으로 매핑되고, 일치하지 않으면 MethodArgumentTypeMismatchException을 발생시키도록 하였습니다.

gachaemin commented on src/main/java/com/playus/communityservice/domain/post/service/PostService.java:
```diff
@@ -0,0 +1,86 @@
+            } catch (NumberFormatException e) {
+                throw new IllegalArgumentException("Image field must be a valid numeric string (file ID)");
+            }
+        }
```
@coderabbitai
형식을 통일하게 변경하겠습니다.

gachaemin commented on src/main/java/com/playus/communityservice/domain/post/service/PostService.java:
```diff
@@ -0,0 +1,86 @@
+        }
+
+        post.delete();
+    }
```
@coderabbitai
파일 전체를 수정하였습니다.

gachaemin commented on src/main/java/com/playus/communityservice/domain/post/service/PostService.java:
```diff
@@ -0,0 +1,86 @@
+        );
+
+        return PostUpdateResponse.ok();
+    }
```
@coderabbitai
위와 같이 수정하였습니다.

gachaemin commented on src/main/java/com/playus/communityservice/global/config/jwt/JwtUser.java:
```diff
@@ -0,0 +1,42 @@
+    private final String role;
+
+    @Override
+    public Collection<? extends GrantedAuthority> getAuthorities() {
+        return null;
+    }
```
@coderabbitai
role자체가 많이 중요하지 않아서 null로 비워놨습니다.

gachaemin commented on src/main/java/com/playus/communityservice/global/config/jwt/JwtFilter.java:
```diff
@@ -0,0 +1,57 @@
+        if (header != null && header.startsWith("Bearer ")) {
+            String token = header.substring(7);
+            try {
+                if (!jwtUtil.isExpired(token)) {
+                    Long userId = Long.parseLong(jwtUtil.getUserId(token));
+                    String role = jwtUtil.getRole(token);
+
+                    JwtUser jwtUser = new JwtUser(userId, role);
+                    Authentication auth = new UsernamePasswordAuthenticationToken(
+                            jwtUser, null, jwtUser.getAuthorities()
+                    );
+                    SecurityContextHolder.getContext().setAuthentication(auth);
+                }
```
@coderabbitai
수정하겠습니다.

gachaemin commented on src/main/java/com/playus/communityservice/global/config/jwt/JwtFilter.java:
```diff
@@ -0,0 +1,57 @@
+                    );
+                    SecurityContextHolder.getContext().setAuthentication(auth);
+                }
+            } catch (JwtException | IllegalArgumentException e) {
+                log.debug("JWT 인증 실패: {}", e.getMessage());
+            }
```
@coderabbitai
이 부분은 이후 추가로 수정할 예정입니다.

gachaemin commented on src/main/java/com/playus/communityservice/domain/post/service/PostService.java:
```diff
@@ -0,0 +1,58 @@
+
+        post.updateAll(request.title(), request.content(), TeamTag.DOOSAN_BEARS, false, request.jwpDate());
+        return PostUpdateResponse.fo(true, "게시물이 수정되었습니다.");
+    }
```
@coderabbitai
수정하겠습니다.

gachaemin commented on src/main/java/com/playus/communityservice/domain/post/controller/PostController.java:
```diff
@@ -0,0 +1,50 @@
+                                                     @AuthenticationPrincipal JwtUser user) {
+        PostUpdateResponse response = postService.updatePost(request, user, tag);
+        return ResponseEntity.ok(response);
+    }
```
@coderabbitai
바로 반영하겠습니다.

gachaemin commented on src/main/java/com/playus/communityservice/domain/post/controller/PostController.java:
```diff
@@ -0,0 +1,50 @@
+        PostCreateResponse response = postService.createPost(request, user, tag);
+        URI location = URI.create("/post/" + response.postId());
+        return ResponseEntity.created(location).body(response);
+    }
```
@coderabbitai
바로 반영하겠습니다.

gachaemin commented on src/main/java/com/playus/communityservice/global/exception/ExceptionAdvice.java:
```diff
@@ -0,0 +1,64 @@
+})
+public class ExceptionAdvice {
+
+    @ResponseStatus(HttpStatus.BAD_REQUEST)
+    @ExceptionHandler(BindException.class)
+    public ResponseEntity<String> bindExceptionHandler(BindException e) {
+        String errorMessage = e.getAllErrors().get(0).getDefaultMessage();
+        log.warn("Validation Error: {}", errorMessage);
+        return ResponseEntity.badRequest().body(errorMessage);
+    }
```
@coderabbitai
수정하도록 하겠습니다.

gachaemin commented on src/main/java/com/playus/communityservice/global/exception/ExceptionAdvice.java:
```diff
@@ -0,0 +1,64 @@
+        return ErrorResponse.forbiddenError(errorMessage);
+    }
+
+    @ResponseStatus(HttpStatus.INTERNAL_SERVER_ERROR)
+    @ExceptionHandler(Exception.class)
+    public ResponseEntity<String> otherExceptionHandler(Exception e) {
+        String errorMessage = e.getMessage();
+        log.error("Unexpected Error: {}", errorMessage);
+        return ResponseEntity.internalServerError().body("서버 에러가 발생했습니다! 관리자에게 문의해 주세요!");
+    }
```
@coderabbitai
수정하겠습니다.

gachaemin committed:
Style: 오타 수정 및 일관성 통일
